pkg/resolver/service/service.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package service
+
+import (
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/resolver"
+	"github.com/ethersphere/bee/pkg/resolver/client/ens"
+)
+
+// InitMultiResolver will create a new MultiResolver, create the appropriate
+// resolvers, push them to the resolver chains and attempt to connect.
+func InitMultiResolver(logger logging.Logger, cfgs []*resolver.ConnectionConfig) *resolver.MultiResolver {
```

Comment:
This looks to me like a constructor for MultiResolver. I think that it can be in the resolver package, where MultiResolver and ConnectionConfig are. And that service package is not needed.

Reply:
Personally, I like that there is a separate factory-like component as it decouples the MultiResolver logic from eg. mapping of domains to resolver clients, parsing connection strings etc.

I believe it breaches the scope of this PR. I will create a separate refactor issue for the resolver.

Reply:
I do not see that what of decoupling is done, as mapping of resolver clients is the responsibility of multiresolver, and parsing connection strings is already in the resolver package ParseConnectionStrings. This function just uses implementations from there. At least do not call it `service` as it does not contain any services, just one constructor function.